server/src/services/metadata.service.ts

```diff
@@ -487,6 +496,124 @@ export class MetadataService implements OnEvents {
     }
   }
 
+  async searchPerson(ownerId: AssetEntity['ownerId'], dto: SearchPeopleDto): Promise<PersonResponseDto[]> {
+    return this.personRepository.getByName(ownerId, dto.name, { withHidden: dto.withHidden });
+  }
+
+  private async applyTaggedFaces(asset: AssetEntity, tags: ImmichTags) {
+    this.logger.debug(`Starting faces extraction from tags (${asset.id})`);
+
+    const faceSet = new Set();
+
+    if (!tags.RegionInfo) {
+      return true;
+    }
+
+    interface Region {
+      imageWidth: number;
+      imageHeight: number;
+      boundingBoxX1: number;
+      boundingBoxX2: number;
+      boundingBoxY1: number;
+      boundingBoxY2: number;
+    }
+
+    interface MetadataFaceResult {
+      Name?: string;
+      Type?: string;
+      Region: Region;
+    }
```

Comment:
> * Types and interfaces should go in `metadata.interface.ts`.

Moved!

>     * `Region` should be more descriptive, maybe `MetadataFaceRegion`?

I'll use MetadataRegion since not all regions in Exif are faces

>     * I personally much prefer `x1`, `y1` etc. to `boundingBoxX1` and want to change that in the DB at some point, so it'd be nice to use that naming here as well. There's [this](https://github.com/immich-app/immich/blob/7c3326b6620ef9e38b1be57f23954b5ef07751ed/server/src/interfaces/machine-learning.interface.ts#L3) interface you can make use of.

Agreed

Reply:
Done
- https://github.com/bugfest/immich/commit/c9c85e115d5ebc2d46fcc80d474ae8736b0810d9#diff-fdc4b3405be1ce95ea9bd5877ce3fae271659bd04481820d6a7bfff09c8b73caR52-R66
- https://github.com/bugfest/immich/commit/c9c85e115d5ebc2d46fcc80d474ae8736b0810d9#diff-61222a9e424956861d68b088bbb08a464f1fa5d8a8c8fa7a3cfa116ae1f8170fR504-R513